### PR TITLE
Section 2.1.3. Extensions

### PR DESCRIPTION
**Pull request by jurij-jukic**

### CODE REVIEW

File: src/process/extensions.md [hunk 1]
```diff
@@ -41,18 +41,20 @@ Check out the [examples below](#examples) to see some working extensions.
 The process [binds a WebSocket](#bind-an-extension-websocket), so Kinode acts as the WebSocket server.
 The extension acts as a client, connecting to the WebSocket served by the Kinode process.
 
-The process sends `HttpServerAction::WebSocketExtPushOutgoing` Requests to the `http_server` to communicate with the extension (see the `enum` defined at the bottom of this section).
+The process sends [`HttpServerAction::WebSocketExtPushOutgoing`](https://github.com/kinode-dao/kinode/blob/main/lib/src/http/server_types.rs#L123-L142) Requests to the `http_server`(look [here](https://book.kinode.org/http_server_and_client.html
+) and [here](https://book.kinode.org/apis/http_server.html)) to communicate with the extension (see the `enum` defined at the bottom of this section).
 
+Table 1: `HttpServerAction::WebSocketExtPushOutgoing` Inputs
```
Comment (nick1udwig) [inline src/process/extensions.md:47]: image

Here and for Table below, need to fix the label. Probably just need to add a newline between label and start of table.

In general, please do check that your changes look good using
```
mdbook serve
```

Check the README on this repo for instructions on getting the required commandline tools.

**Comment by jurij-jukic** [comment]
removed github links prone to deprecation, will add process_lib links when it get synced

File: src/process/extensions.md [hunk 2]
```diff
@@ -78,7 +80,7 @@ pub enum HttpServerAction {
     /// * blob as given.
     WebSocketExtPushData {
         id: u64,
-        kinode_message_type: MessageType,
+        kinode_message_type: Message,
```
[the remark this answers is not in the document]
Comment (jurij-jukic) [reply src/process/extensions.md:83]: my bad. i was looking at the process_lib docs and couldn't find `MessageType` so i assumed it was a typo here

File: src/process/extensions.md [hunk 3]
```diff
@@ -87,12 +89,15 @@ pub enum HttpServerAction {
 
 ### The Package
 
-The package is, minimally, a single process that serves as interface between Kinode and the extension.
+The package is, minimally, a single process that serves as interface between Kinode and the extension. Each extension must come with a corresponding Kinode package.
```
[the remark this answers is not in the document]
Comment (jurij-jukic) [reply src/process/extensions.md:92]: gotcha, this one slipped

File: src/process/extensions.md (hunk 1 above)
Comment (jurij-jukic) [reply src/process/extensions.md:47]: gotcha, i was using a markdown preview extension in vscode, but seems that everything renders md differently

**Pull request merged**